src/file-manager/fm-icon-view.c

```diff
@@ -990,6 +996,38 @@ fm_icon_view_set_directory_keep_aligned (FMIconView *icon_view,
      keep_aligned);
 }
 
+static gboolean
+get_default_directory_lock_icons_position (void)
+{
+    return TRUE;
+}
+
+static gboolean
+fm_icon_view_get_directory_lock_icons_position (FMIconView *icon_view,
+        CajaFile *file)
+{
+    if (!fm_icon_view_supports_keep_aligned (icon_view))
+    {
+        return FALSE;
+    }
+
+    return caja_file_get_boolean_metadata
+            (file,
+             CAJA_METADATA_KEY_ICON_VIEW_LOCK_ICONS_POSITION,
+             get_default_directory_lock_icons_position ());
+}
+
+static void
+fm_icon_view_set_directory_lock_icons_position (FMIconView *icon_view,
```

Comment:
Warning.
https://travis-ci.org/github/mate-desktop/caja/jobs/764131695#L11413
```
fm-icon-view.c:1021:61: warning: unused parameter 'icon_view' [-Wunused-parameter]

fm_icon_view_set_directory_lock_icons_position (FMIconView *icon_view,
```

Reply:
Fair enough; what am I supposed to do here?  I recall last time I used `G_GNUC_UNUSED` it started a abundant thread on how we should suppress that, so I'd rather not replicate that here and use whatever is preferred :slightly_smiling_face:

Reply:
Actually it could make sense to test the view settings, so I'll do and use that parameter instead.